planning/behavior_path_planner_common/src/utils/drivable_area_expansion/drivable_area_expansion.cpp

```diff
@@ -121,52 +121,79 @@ double calculate_minimum_lane_width(
+  bound_projections.resize(path_poses.size(), {{}, std::numeric_limits<double>::max()});
   for (auto path_idx = 0UL; path_idx < path_poses.size(); ++path_idx) {
-    const auto & path_pose = path_poses[path_idx];
-    if (curvatures[path_idx] == 0.0) continue;
-    const auto curvature_radius = 1 / curvatures[path_idx];
-    const auto min_lane_width = calculate_minimum_lane_width(curvature_radius, params);
-    const auto side_distance = min_lane_width / 2.0 * (side == LEFT ? 1.0 : -1.0);
-    const auto offset_point =
-      tier4_autoware_utils::calcOffsetPose(path_pose, 0.0, side_distance, 0.0).position;
+    const auto path_p = convert_point(path_poses[path_idx].position);
     for (auto bound_idx = lb_idx; bound_idx + 1 < bound.size(); ++bound_idx) {
-      const auto & prev_p = bound[bound_idx];
-      const auto & next_p = bound[bound_idx + 1];
-      const auto intersection_point =
-        tier4_autoware_utils::intersect(offset_point, path_pose.position, prev_p, next_p);
-      if (intersection_point) {
-        lb_idx = bound_idx;
-        const auto dist = tier4_autoware_utils::calcDistance2d(*intersection_point, offset_point);
-        minimum_expansions[bound_idx] = std::max(minimum_expansions[bound_idx], dist);
-        minimum_expansions[bound_idx + 1] = std::max(minimum_expansions[bound_idx + 1], dist);
-        // apply the expansion to all bound points within the extra arc length
-        auto arc_length =
-          tier4_autoware_utils::calcDistance2d(*intersection_point, bound[bound_idx + 1]);
-        for (auto up_bound_idx = bound_idx + 2; up_bound_idx < bound.size(); ++up_bound_idx) {
-          arc_length +=
-            tier4_autoware_utils::calcDistance2d(bound[up_bound_idx - 1], bound[up_bound_idx]);
-          if (arc_length > params.extra_arc_length) break;
-          minimum_expansions[up_bound_idx] = std::max(minimum_expansions[up_bound_idx], dist);
-        }
-        arc_length = tier4_autoware_utils::calcDistance2d(*intersection_point, bound[bound_idx]);
-        for (auto down_offset_idx = 1LU; down_offset_idx < bound_idx; ++down_offset_idx) {
-          const auto idx = bound_idx - down_offset_idx;
-          arc_length += tier4_autoware_utils::calcDistance2d(bound[idx - 1], bound[idx]);
-          if (arc_length > params.extra_arc_length) break;
-          minimum_expansions[idx] = std::max(minimum_expansions[idx], dist);
-        }
-        break;
+      const auto prev_p = convert_point(bound[bound_idx]);
+      const auto next_p = convert_point(bound[bound_idx + 1]);
+      const auto projection = point_to_segment_projection(path_p, prev_p, next_p);
+      if (projection.distance < bound_projections[path_idx].distance) {
+        bound_indexes[path_idx] = bound_idx;
+        bound_projections[path_idx] = projection;
       }
     }
+    lb_idx = bound_indexes[path_idx];
+  }
+}
+
+void apply_extra_arc_length(
+  Expansion & expansion, const std::vector<Point> & bound, const double extra_arc_length,
```

Comment:
`extra_arc_length`  is confusing. Need a better name.

Reply:
[d0490c4](https://github.com/autowarefoundation/autoware.universe/pull/5640/commits/d0490c461f9926cdbf3be61e252e28af68bf9b5e)